test/FunctionalTests/Microsoft.OData.Client.Tests/DataServiceQueryProviderTests.cs

```diff
@@ -180,13 +180,13 @@ public void TranslatesStaticFunction()
             localDsc.ResolveName = (t) => "ServiceNamespace.Product";
             var sut = new DataServiceQueryProvider(localDsc);
             var products = localDsc.CreateQuery<Product>("Products")
-                .Where(product => Product.StaticFunction(product.Name));
+                .Where(product => true && Product.StaticFunction(product.Name));
 
             // Act
             var queryComponents = sut.Translate(products.Expression);
 
             // Assert
-            Assert.Equal(@"http://root/Products?$filter=ServiceNamespace.StaticFunction(parameter=$it/Name)", queryComponents.Uri.ToString());
+            Assert.Equal(@"http://root/Products?$filter=true and ServiceNamespace.StaticFunction(parameter=$it/Name)", queryComponents.Uri.ToString());

```

Comment:
Why is this change necessary?

Reply:
This is the test part mentioned above that relates to the incomplete fix: https://github.com/OData/odata.net/pull/2572
Adding `true &&` will make parent not null in ExpressionWriter VisitMethodCall, but there havent been written anything in front of the slash.

Reply:
Could this be separate test to ensure that both versions are supported?

Reply:
Sure. I will add it as a new test instead.

Reply:
Pushed change. Thanks.